### PR TITLE
Improved code documentation in resources/views.py

### PR DESCRIPTION
**Pull request by Samwalton9**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Ran through the entire file with fairly fresh eyes, documenting anything that was initially unclear to me.

I also refactored one piece of code where we were duplicating lines a bunch of times, adding a couple of model properties and a method instead. This could have technically been simplified further, but I tried to keep things slightly generic in case, for example, we just want to fetch a partner's short description.

## Phabricator Ticket
https://phabricator.wikimedia.org/T264424

## How Has This Been Tested?
Poked around the site locally, and ran tests.

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/resources/views.py [hunk 1]
```diff
@@ -73,28 +73,32 @@ def get_context_data(self, **kwargs):
...
             self.request.GET, queryset=self.get_queryset(), language_code=language_code
         )
         context["filter"] = partner_filtered_list
 
+        # Retrieve the current user and add to context
         user = self.request.user
         if user.is_authenticated:
             user = User.objects.select_related("editor").get(pk=self.request.user.pk)
             context["user"] = user
-            context["editor"] = user.editor
```
Comment (Samwalton9) [inline TWLight/resources/views.py:94]: This didn't seem to be used anywhere. Replaced it with a string and couldn't find the string in page source, and couldn't find `editor` in the relevant html files. Nothing seemed to break - it looks like we're accessing the `editor` via `user.editor` in the html.

File: TWLight/resources/views.py [hunk 2]
```diff
@@ -36,65 +34,70 @@
...
-                .order_by("company_name")
-            )
+            qs = Partner.even_not_available.prefetch_related("languages")
+        else:
+            qs = Partner.objects.prefetch_related("languages")
 
-        return qs
+        # The ordering here is useful primarily to people familiar with the
+        # English alphabet. :/
+        partner_queryset = qs.select_related("coordinator", "logos").order_by(
```
Comment (jsnshrmn) [inline TWLight/resources/views.py:63]: This is a big improvement; I think you could put the prefetch here to, for even more DRYness.

**Review by jsnshrmn — COMMENTED** [review]
This looks good and worked as expected in my local testing. I have one non-blocking suggestion. Feel free to decline it if you don't like it! Either way, I'll happily merge.

File: TWLight/resources/views.py (hunk 2 above)
Comment (Samwalton9) [reply TWLight/resources/views.py]: Done!

**Review by jsnshrmn — APPROVED** [review]
Thanks for your work on this!

**Pull request merged**